### ISSUE TITLE
Hanging on huge tests in 4.0.0b3

### ISSUE THREAD

**Issue by arogozhnikov** [cupy/cupy#939]
Hi, got on to testing of 4.0.0b3 with docker. 

One of the tests hangs deadly, namely `test_cumprod_huge_array`

Here is a small snippet that is enough to reproduce

```python
size = 2 ** 32 # works fine for 2 ** 31
a = cupy.ones(size, 'b')
print(a[0])
```

Do you have ideas about the reasons?

**Comment by arogozhnikov** [comment]
Forgot to mention: I have 12GB on GPU, which should be enough. Memory is allocated, so the possible problem is somewhat connected to indices types (int32 / uint32).

**Comment by kmaehashi** [comment]
Thanks for reporting! The issue will be fixed in the next release (v4.0.0b4 / v2.4.0).
https://github.com/cupy/cupy/pull/882

**Comment by kmaehashi** [comment]
Fixed via #882.
